### ISSUE TITLE
OCS deploy using OCS-CI failed at post deploy validation stage

### ISSUE THREAD

**Issue by gitsridhar** [red-hat-storage/ocs-ci#3283]
Today ppc64le environment: ocs-ci based OCS deploy failed. The error is in the validation step. I see OCS deployed, I see pods in openshift-storage namespace. 

Command used to deploy OCS 👍 
```
run-ci -m deployment --deploy \
        --ocs-version 4.6 --cluster-name=ocstest \
        --ocsci-conf=conf/ocsci/production_powervs_upi.yaml \
        --ocsci-conf=$WORKSPACE/ocs-ci-conf.yaml \
        --cluster-path=$WORKSPACE --collect-logs tests/
```
From output:
```
=================================== FAILURES ===================================
_______________________________ test_deployment ________________________________

pvc_factory = <function pvc_factory_fixture.<locals>.factory at 0x200010ac3488>
pod_factory = <function pod_factory_fixture.<locals>.factory at 0x200010ac3510>

    @deployment
    @polarion_id(get_polarion_id())
    def test_deployment(pvc_factory, pod_factory):
        deploy = config.RUN['cli_params'].get('deploy')
        teardown = config.RUN['cli_params'].get('teardown')
        if not teardown or deploy:
            log.info("Verifying OCP cluster is running")
            assert is_cluster_running(config.ENV_DATA['cluster_path'])
            if not config.ENV_DATA['skip_ocs_deployment']:
                ocs_registry_image = config.DEPLOYMENT.get(
                    'ocs_registry_image'
                )
>               ocs_install_verification(ocs_registry_image=ocs_registry_image)

tests/ecosystem/deployment/test_deployment.py:25:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
ocs_ci/ocs/resources/storage_cluster.py:290: in ocs_install_verification
    validate(instance=item, schema=schemas[item['type']])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

instance = {'crush_weight': 0.25, 'depth': 3, 'device_class': 'hdd', 'exists': 1, ...}
schema = {'additionalProperties': False, 'properties': {'crush_weight': {'type': 'number'}, 'depth': {'type': 'integer'}, 'devi...'integer'}, ...}, 'required': ['crush_weight', 'depth', 'device_class', 'exists', 'id', 'name', ...], 'type': 'object'}
cls = <class 'jsonschema.validators.create.<locals>.Validator'>, args = ()
kwargs = {}
validator = <jsonschema.validators.create.<locals>.Validator object at 0x20000a199a20>
error = <ValidationError: "'up' was expected">

    def validate(instance, schema, cls=None, *args, **kwargs):
        """
        Validate an instance under the given schema.

            >>> validate([2, 3, 4], {"maxItems": 2})
            Traceback (most recent call last):
                ...
            ValidationError: [2, 3, 4] is too long
```

**Comment by gitsridhar** [comment]
Latest run has a different error:
self = <ocs_ci.ocs.resources.packagemanifest.PackageManifest object at 0x2000108c4470>
kwargs = {}, selector = 'ocs-operator-internal=true'
data = {'apiVersion': 'v1', 'items': [], 'kind': 'List', 'metadata': {'resourceVersion': '', 'selfLink': ''}}
items = [], data_len = 0

    def get(self, **kwargs):
        """
        Overloaded get method from OCP class.

        Raises:
            ResourceNotFoundError: In case the selector and resource_name
                specified and no such resource found.
        """
        resource_name = kwargs.get("resource_name", "")
        resource_name = resource_name if resource_name else self.resource_name
        selector = kwargs.get('selector')
        selector = selector if selector else self.selector

        data = super(PackageManifest, self).get(**kwargs)
        if type(data) == dict and (data.get('kind') == 'List'):
            items = data['items']
            data_len = len(items)
            if data_len == 0 and selector and resource_name:
                raise ResourceNotFoundError(
>                   f"Requested packageManifest: {resource_name} with "
                    f"selector: {selector} not found!"
                )
E               ocs_ci.ocs.exceptions.ResourceNotFoundError: Requested packageManifest: ocs-operator with selector: ocs-operator-internal=true not found!

ocs_ci/ocs/resources/packagemanifest.py:72: ResourceNotFoundError

**Comment by clacroix12** [comment]
The first issue is a failure during validation of the ceph osd tree output, maybe @shylesh can weigh in on why this is failing.

The second issue has been reported here: https://github.com/red-hat-storage/ocs-ci/issues/3285

**Comment by zmc** [comment]
I haven't seen much output from these validators, but it looks like one of your OSDs didn't come up:
`error = <ValidationError: "'up' was expected">`

I'd take a look at the OSDs to see if that's true, and why.

**Comment by gitsridhar** [comment]
The OSDs are fine. but even then, seeing the post validation error:
rook-ceph-osd-0-6bf7ff5595-s2vjw                                  1/1     Running     0          15m
rook-ceph-osd-1-678dc6b9-lr5rx                                    1/1     Running     0          14m
rook-ceph-osd-2-69797494fd-9qv2t                                  1/1     Running     0          17m

Error:

=================================== FAILURES ===================================
_______________________________ test_deployment ________________________________

pvc_factory = <function pvc_factory_fixture.<locals>.factory at 0x2000118a19d8>
pod_factory = <function pod_factory_fixture.<locals>.factory at 0x2000118a1840>

    @deployment
    @polarion_id(get_polarion_id())
    def test_deployment(pvc_factory, pod_factory):
        deploy = config.RUN['cli_params'].get('deploy')
        teardown = config.RUN['cli_params'].get('teardown')
        if not teardown or deploy:
            log.info("Verifying OCP cluster is running")
            assert is_cluster_running(config.ENV_DATA['cluster_path'])
            if not config.ENV_DATA['skip_ocs_deployment']:
                ocs_registry_image = config.DEPLOYMENT.get(
                    'ocs_registry_image'
                )
>               ocs_install_verification(ocs_registry_image=ocs_registry_image)

tests/ecosystem/deployment/test_deployment.py:25:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
ocs_ci/ocs/resources/storage_cluster.py:290: in ocs_install_verification
    validate(instance=item, schema=schemas[item['type']])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

instance = {'crush_weight': 0.25, 'depth': 3, 'device_class': 'hdd', 'exists': 1, ...}
schema = {'additionalProperties': False, 'properties': {'crush_weight': {'type': 'number'}, 'depth': {'type': 'integer'}, 'devi...'integer'}, ...}, 'required': ['crush_weight', 'depth', 'device_class', 'exists', 'id', 'name', ...], 'type': 'object'}
cls = <class 'jsonschema.validators.create.<locals>.Validator'>, args = ()
kwargs = {}
validator = <jsonschema.validators.create.<locals>.Validator object at 0x200010943198>
error = <ValidationError: "'up' was expected">

    def validate(instance, schema, cls=None, *args, **kwargs):
        """
        Validate an instance under the given schema.

            >>> validate([2, 3, 4], {"maxItems": 2})
            Traceback (most recent call last):
                ...
            ValidationError: [2, 3, 4] is too long

        :func:`validate` will first verify that the provided schema is
        itself valid, since not doing so can lead to less obvious error
        messages and fail in less obvious or consistent ways.

        If you know you have a valid schema already, especially if you
        intend to validate multiple instances with the same schema, you
        likely would prefer using the `IValidator.validate` method directly
        on a specific validator (e.g. ``Draft7Validator.validate``).


        Arguments:

            instance:

                The instance to validate

            schema:

                The schema to validate with

            cls (IValidator):

                The class that will be used to validate the instance.

        If the ``cls`` argument is not provided, two things will happen
        in accordance with the specification. First, if the schema has a
        :validator:`$schema` property containing a known meta-schema [#]_
        then the proper validator will be used. The specification recommends
        that all schemas contain :validator:`$schema` properties for this
        reason. If no :validator:`$schema` property is found, the default
        validator class is the latest released draft.

        Any other provided positional and keyword arguments will be passed
        on when instantiating the ``cls``.

        Raises:

            `jsonschema.exceptions.ValidationError` if the instance
                is invalid

            `jsonschema.exceptions.SchemaError` if the schema itself
                is invalid

        .. rubric:: Footnotes
        .. [#] known by a validator registered with
            `jsonschema.validators.validates`
        """
        if cls is None:
            cls = validator_for(schema)

        cls.check_schema(schema)
        validator = cls(schema, *args, **kwargs)
        error = exceptions.best_match(validator.iter_errors(instance))
        if error is not None:
>           raise error
E           jsonschema.exceptions.ValidationError: 'up' was expected
E
E           Failed validating 'const' in schema['properties']['status']:
E               {'const': 'up'}
E
E           On instance['status']:
E               'down'

../../../venv/lib64/python3.6/site-packages/jsonschema/validators.py:934: ValidationError
---------------------------- Captured stdout setup -----------------------------
collected 579 items / 570 deselected / 9 selected

tests/ecosystem/deployment/test_deployment.py::test_deployment
--------------------------- Captured stdout teardown ---------------------------
FAILED                                                                   [100%]
=============================== warnings summary ===============================
/home/test/venv/lib64/python3.6/site-packages/_pytest/config/__init__.py:891
/home/test/venv/lib64/python3.6/site-packages/_pytest/config/__init__.py:891
  /home/test/venv/lib64/python3.6/site-packages/_pytest/config/__init__.py:891: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: pytest_reportportal
    self._mark_plugins_for_rewrite(hook)

/home/test/venv/lib64/python3.6/site-packages/_pytest/mark/structures.py:327
  /home/test/venv/lib64/python3.6/site-packages/_pytest/mark/structures.py:327: PytestUnknownMarkWarning: Unknown pytest.mark.post_ocp_upgrade - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

/home/test/venv/lib64/python3.6/site-packages/_pytest/mark/structures.py:327
  /home/test/venv/lib64/python3.6/site-packages/_pytest/mark/structures.py:327: PytestUnknownMarkWarning: Unknown pytest.mark.last - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

-- Docs: https://docs.pytest.org/en/latest/warnings.html
========== 1 failed, 570 deselected, 4 warnings in 1189.47s (0:19:49) ==========

Not happening consistently every time though.

**Comment by shylesh** [comment]
@gitsridhar , As Zack pointed out it something wrong with osd status. Please check 'ceph osd tree'  and make sure that all the osds are in 'up' state. There is mismatch between schema and expected output.

schema is 
==================
OSD_TREE_OSD = {
    "type": "object",
    "properties": {
        "id": {"type": "integer"},
        "device_class": {"type": "string"},
        "name": {"pattern": "osd[.][0-9]+"},
        "type": {"const": "osd"},
        "type_id": {"const": 0},
        "crush_weight": {"type": "number"},
        "depth": {"type": "integer"},
        "pool_weights": {"type": "object"},
        "exists": {"type": "integer"},
        "status": {"const": "up"},
        "reweight": {"type": "integer"},
        "primary_affinity": {"type": "integer"},
    },
    "required": [
        "crush_weight",
        "depth",
        "device_class",
        "exists",
        "id",
        "name",
        "pool_weights",
        "primary_affinity",
        "reweight",
        "status",
        "type",
        "type_id",
    ],
    "additionalProperties": False,
}


=================

Here status expected to be 'up' but looks like in your case its in 'down' state.

**Comment by gitsridhar** [comment]
Right after the validation failure, I ran this:
[test@osp121 logs]$ oc -n openshift-storage rsh rook-ceph-tools-85b9f46dc8-p7zpj ceph osd tree
ID  CLASS WEIGHT  TYPE NAME             STATUS REWEIGHT PRI-AFF 
 -1       0.75000 root default                                  
 -8       0.25000     rack rack0                                
 -7       0.25000         host worker-0                         
  2   hdd 0.25000             osd.2         up  1.00000 1.00000 
 -4       0.25000     rack rack1                                
 -3       0.25000         host worker-1                         
  0   hdd 0.25000             osd.0         up  1.00000 1.00000 
-12       0.25000     rack rack2                                
-11       0.25000         host worker-2                         
  1   hdd 0.25000             osd.1         up  1.00000 1.00000 
[test@osp121 logs]

I see all osds up.

After this step in ocs validation step, must-gather was running for quite a long time - 5 minutes or so and then I executed above command. I think the failing validation step needs to wait bit longer.
